src/test/java/itest/CustomTargetsIT.java

```diff
@@ -341,9 +349,11 @@ void shouldBeAbleToDeleteTarget()
                     }
                 });
 
+        String jmxServiceUrl = "service:jmx:rmi:///jndi/rmi://localhost:0/jmxrmi";
+        String encodedUrl = URLEncoder.encode(jmxServiceUrl, StandardCharsets.UTF_8);
```

Comment:
Despite its name, URLEncoder is not actually the right tool for this job:

https://docs.oracle.com/javase/8/docs/api/java/net/URLEncoder.html

> Utility class for HTML form encoding. This class contains static methods for converting a String to the application/x-www-form-urlencoded MIME format.

I think it's better to use URLEncodedUtils:

https://github.com/cryostatio/cryostat/blob/c2b26e1019c3852bea0436fc8dfef1e951b78c7a/src/test/java/itest/bases/StandardSelfTest.java#L47

https://hc.apache.org/httpcomponents-client-4.5.x/current/httpclient/apidocs/org/apache/http/client/utils/URLEncodedUtils.html#formatSegments(java.lang.Iterable,%20java.nio.charset.Charset)